core/autoscaling.go

```diff
@@ -536,6 +595,18 @@ func (a *autoScalingGroup) getAllowedInstanceTypes(baseInstance *instance) []str
 
 }
 
+func (a *autoScalingGroup) getPricetoBid(
+	baseOnDemandPrice float64, currentSpotPrice float64) float64 {
+
+	if a.SpotPriceMultiplierFlag == true {
+		logger.Println("Launching spot instance with a bid =", currentSpotPrice*a.region.conf.SpotPriceMultiplier)
+		return currentSpotPrice * a.region.conf.SpotPriceMultiplier
+	}
+
+	logger.Println("Launching spot instance with a bid =", baseOnDemandPrice)
+	return baseOnDemandPrice
+}
+
```

Comment:
This method would be obsolete if the default parameter was added and the flag removed. The multiplier could then either be used here: https://github.com/cristim/autospotting/blob/master/core/region.go#L236
or here: https://github.com/cristim/autospotting/blob/master/core/instance.go#L160-L162

Just like this new parameter actually: https://github.com/cristim/autospotting/pull/163/files#diff-17a0faaa6296d309bb099b250ff6ead1R162.

I'd tend to prefer putting them in the `instance.go` file as it seems more logical, but the main method would need to take new parameters such as `biding_policies` or `multipliers` or something like this.

@cristim what are your thoughts on this?

Reply:
It can be done, but at the repo level - requestSpotPrices() might be used for another bidding policy in the future and I wouldn't want to touch that. isPriceCompatible() is used to determine the choice of spot instances to be spawned for the current on-demand instance - Also wouldn't want to change that. I did think of these scenarios before I begun the implementation, but I figured it would be best placed in autoscaling.go since that is where you call for the bidPrice to be placed.